### PR TITLE
Parameterize apt's "cache_valid_time" when installing "fail2ban"

### PR DESCRIPTION
**Pull request by tehranian**

### CODE REVIEW

**Comment by tersmitten** [comment]
Why do you want to change the cache_valid_time?

**Comment by tehranian** [comment]
Hi @tersmitten,

I want to parameterize the `cache_valid_time` in order to help speed up our Ansible playbook runs. We have many nodes which, in-turn, have many apt repos that they pull from. Despite local apt caching proxies, the process of running "apt-get update" on these nodes is _extremely_ time consuming.

Ultimately, these "apt-get update"s are usually completely unnecessary for our needs. For our infra we don't need up to the hour/minute metadata about the latest apt packages. Daily "apt-get update"s are just fine for our needs. Parameterizing this gives us the ability to control that. (We're using a Jenkins job with a string parameter to override via Ansible's `--extra-args="..."` flag)

Ex: After going through all of our "apt" commands and removing the unnecessary "apt-get update"s from our playbooks and 3rdparty modules, we have our playbook run down from ~60 mins -> 7 mins (!!)

I've had similar pull requests accepted to other Ansible modules. Ex: https://github.com/angstwad/docker.ubuntu/commit/9a23673dd793f984850079a08cc2ae505fa2eaf6

Thanks!
Dan

**Comment by tersmitten** [comment]
I understand. ~60 mins -> 7 mins is a real time saver. The problem is that we have a lot of roles and I rather have one place to override `cache_valid_time`.

I was thinking of using [this variable](https://github.com/Oefenweb/ansible-apt/blob/master/defaults/main.yml#L21) from our [apt role](https://github.com/Oefenweb/ansible-apt) and when that variable is undefined defaulting to 3600. What do you think about that? Or do you have any other recommendations?

**Comment by tehranian** [comment]
Hi @tersmitten ,

Sure, using that variable name instead works for me. Want me to send a new PR?

Thanks.

File: tasks/main.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     name: fail2ban
     state: latest
     update_cache: true
-    cache_valid_time: 3600
+    cache_valid_time: "{{ fail2ban_apt_cache_valid_time }}"
```
Comment (tersmitten) [inline tasks/main.yml]: Change this to `cache_valid_time: "{{ apt_update_cache_valid_time | default(3600) }}"`

File: defaults/main.yml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # defaults file for fail2ban
 ---
+fail2ban_apt_cache_valid_time: 3600
```
Comment (tersmitten) [inline defaults/main.yml]: Can be removed

File: README.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ Set up fail2ban in Debian-like systems.
 None
 
 #### Variables
+- `fail2ban_apt_cache_valid_time`: [default: `3600`]: Time (in seconds) to pass to Ansible for updating the apt-cache before installing fail2ban
```
Comment (tersmitten) [inline README.md]: Can be removed

**Comment by tersmitten** [comment]
I added some notes. Please update your pull request so I can merge it (and apply it to all our other roles:-))

**Comment by tehranian** [comment]
Closing this in favor of the change in #14.

Thanks,
Dan

**Pull request closed without merging**